### PR TITLE
batch-changes: ensure PR is only approved after code owner review

### PR DESCRIPTION
**Pull request by davejrt**
Attempts to address https://github.com/sourcegraph/sourcegraph/issues/50874

By getting the "ReviewDecision" from a PR which is only "APPROVED" once a codeowner (if required) has reviewed the PR, otherwise this field returns "REVIEW_REQUIRED" or "CHANGES_REQUSTED".

Given there is a no longer a requirement for events on github PRs, we don't need to calculate if the changeset is older than events before computing the state. 

## Test plan

CI passing, tested locally with a repo on ghe.sgdev.org with a codeowners file

### CODE REVIEW

**Comment by davejrt** [comment]
Current issue is, the MergableState field on the pullrequest type is empty, despite it being populated when querying the pull request via the github api:

```
  "BaseRepository": {
    "ID": "MDEwOlJlcG9zaXRvcnk5Nzg=",
    "Name": "JetBrains-ideavim",
    "Owner": {
      "Login": "sourcegraph"
    }
  },
  "HeadRepository": {
    "ID": "MDEwOlJlcG9zaXRvcnk5Nzg=",
    "Name": "JetBrains-ideavim",
    "Owner": {
      "Login": "sourcegraph"
    }
  },
  "MergeableState": ""
}
```
```
  "author_association": "MEMBER",
  "auto_merge": null,
  "active_lock_reason": null,
  "merged": false,
  "mergeable": true,
  "rebaseable": true,
  "mergeable_state": "blocked",
  "merged_by": null,
  "comments": 0,
  "review_comments": 0,
  "maintainer_can_modify": false,
  "commits": 1,
  "additions": 1,
  "deletions": 0,
  "changed_files": 1
}```

**Comment by davejrt** [comment]
Example of what is returned from [createPullRequest](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/extsvc/github/common.go?L610:20-610:37) where mergablestate is empty

```json
pr: &{RepoWithOwner: 
ID:MDExOlB1bGxSZXF1ZXN0OTg1 
Title:Hello World 
Body:My first batch change!
[_Created by Sourcegraph batch change `admin/hello-world`._]
(https://sourcegraph.test:3443/users/admin/batch-changes/hello-world) 
State:OPEN URL:https://ghe.sgdev.org/sourcegraph/JetBrains-ideavim/pull/15 
HeadRefOid:d08b7c73844d4d97beafdba5def4b0cea6fd1639 
BaseRefOid:e44ea7ae691ee9004d0b0c9167a58af044e94ba9 
HeadRefName:hello-world-2 
BaseRefName:master 
Number:15 
MergeableState: 
Author:{AvatarURL:https://ghe.sgdev.org/avatars/u/3 
Login:milton 
URL:https://ghe.sgdev.org/milton} 
BaseRepository:{ID:MDEwOlJlcG9zaXRvcnk5Nzg= 
Name:JetBrains-ideavim Owner:{Login:sourcegraph}} 
HeadRepository:{ID:MDEwOlJlcG9zaXRvcnk5Nzg= 
Name:JetBrains-ideavim 
Owner:{Login:sourcegraph}} 
Participants:[] 
Labels:{Nodes:[]} 
TimelineItems:[] 
Commits:{Nodes:[{Commit:{OID:d08b7c73844d4d97beafdba5def4b0cea6fd1639 
CheckSuites:{Nodes:[]} 
Status:{State: 
Contexts:[]} 
CommittedDate:2023-06-16 15:00:13 +0000 UTC}}]} 
IsDraft:false 
CreatedAt:2023-06-16 15:00:33 +0000 UTC 
UpdatedAt:2023-06-16 15:00:33 +0000 UTC}
```

File: enterprise/internal/batches/types/changeset.go [hunk 1]
```diff
@@ -10,10 +10,13 @@ import (
 	"github.com/goware/urlx"
 	"github.com/inconshreveable/log15"
 	"github.com/sourcegraph/go-diff/diff"
+<<<<<<< HEAD
+=======
 	"github.com/sourcegraph/sourcegraph/internal/gitserver/protocol"
 
 	gerritbatches "github.com/sourcegraph/sourcegraph/enterprise/internal/batches/sources/gerrit"
 	"github.com/sourcegraph/sourcegraph/internal/extsvc/azuredevops"
+>>>>>>> main
```
Comment (BolajiOlajide) [inline enterprise/internal/batches/types/changeset.go]: Hey! It seems there are some git markers left over from a rebase operation.
```suggestion
	"github.com/sourcegraph/sourcegraph/internal/gitserver/protocol"

	gerritbatches "github.com/sourcegraph/sourcegraph/enterprise/internal/batches/sources/gerrit"
	"github.com/sourcegraph/sourcegraph/internal/extsvc/azuredevops"
```

File: internal/extsvc/github/common.go [hunk 2]
```diff
@@ -241,6 +241,7 @@ type PullRequest struct {
 	HeadRefName    string
 	BaseRefName    string
 	Number         int64
+	ReviewDecision string
```
Comment (BolajiOlajide) [inline internal/extsvc/github/common.go:244]: I noticed that the `ReviewDecision` field only exists in the GraphQL response for loading a pull request and not the response from the REST API.

CleanShot 2023-06-22 at 12 46 04@2x

I've checked and confirmed we only load pull requests using the v4 client.

**Review by BolajiOlajide — APPROVED** [review]
Thanks for the fix @davejrt !

**Pull request merged**